### DISCUSSION TITLE
DOTA2 resolution is squished, may be due to incorrectly identified primary monitor

### DISCUSSION THREAD

**Discussion by rykugur** [hyprwm/Hyprland#4130] [Q&A] [answered]
Hi there, I'm new to wayland and hyprland, but have been using linux (mostly i3wm) as my daily for a number of years now. I'm liking hyprland a _LOT_, and the only issue I'm having at the moment is that my DOTA2 game window is appearing very squished.

 **Issue**: DOTA2 is "Squished". After some tinkering I discovered that it's defaulting to my secondary monitor and its max  resolution. The game seems to be defaulting to 1080x1440 on a 3440x1440 monitor.

I tested disabling my secondary monitor and was able to get DOTA2 to display the correct resolution and consistently launch with that resolution. It even remained after I re-enabled the secondary monitor. However, the issue was back after a reboot.

**Second issue**: the game flickers quite badly when I'm able to run at full resolution. Other games (E.g. EVE Online) do not seem to have this issue.

I have two monitors:

```
 hyprctl monitors
Monitor HDMI-A-1 (ID 0):
	1920x1080@143.85699 at 3440x0
	description: ASUSTek COMPUTER INC ASUS XG16A 0x000175F5 (HDMI-A-1)
	make: ASUSTek COMPUTER INC
	model: ASUS XG16A
	serial: 0x000175F5
	active workspace: 6 (6)
	special workspace: 0 ()
	reserved: 0 30 0 0
	scale: 1.00
	transform: 1
	focused: no
	dpmsStatus: 1
	vrr: 0
	activelyTearing: false

Monitor DP-2 (ID 1):
	3440x1440@143.97501 at 0x0
	description: Dell Inc. Dell AW3423DW #G7IYMxgwAA8L (DP-2)
	make: Dell Inc.
	model: Dell AW3423DW
	serial: #G7IYMxgwAA8L
	active workspace: 1 (1)
	special workspace: 0 ()
	reserved: 0 30 0 0
	scale: 1.00
	transform: 0
	focused: yes
	dpmsStatus: 1
	vrr: 1
	activelyTearing: false
```

Note: these are technically out or order; my 3440x1440 (DP-2) monitor is my "primary". Could that be the reason? Is there a way to change that?

Both plugged into an Nvidia 4070.

Another note: I've got `1Password` set to `exec-once` on launch; it always starts on my secondary monitor.

Attaching a couple screenshots for reference.

![image](https://github.com/hyprwm/Hyprland/assets/2609898/5b64adc0-9ba1-493e-96d1-bb2f106bd07e)
![image](https://github.com/hyprwm/Hyprland/assets/2609898/290c64f9-ee15-411f-a5ee-fdf175b421d3)

**Comment by zakk4223** [comment, marked as answer]
It's probably the stupid/classic 'xwayland has no primary monitor set' issue. 

Try running 'xrandr --output DP-2 --primary' before you run DOTA and see if that helps.
If it does, consider using https://github.com/zakk4223/hyprXPrimary

**Reply by rykugur** [reply]
That did it! Would have taken me ages to figure that out. Thank you! Definitely gonna look into this hyprXPrimary thing.

Now just need to figure out/fix the flickering and I'm set!
